books/RayTracingTheNextWeek.html

```diff
@@ -1810,7 +1818,7 @@
             auto *bptr = bdata;
             auto *fptr = fdata;
             for (auto i=0; i < total_bytes; i++, fptr++, bptr++)
-                *bptr = static_cast< unsigned char >(*fptr * 256.0);
+                *bptr = float_to_byte(*fptr);
```

Comment:
instead of making a new method, couldn't we just use the clamp defined above?

```
*bptr = static_cast< unsigned char >(clamp(256.0 * value, 0, 255));
```

Reply:
Hmmm. The existing `clamp()` function clamps 32-bit integers. So I could first convert to `int`, though I'm not sure what values less than -2.147483648e9 would convert to, nor what values greater than +2.147483647e9 would convert to. These are susceptible to the same over/underflow results that the 8-bit values displayed. I do know that HDR values are supported, so we're not restricted to [0,1) or [0,1]. Thus, I'd have to perform some clamping _before_ safely passing values to the `int` clamping function, and then once more to clamp _those_ values to the unsigned 8-bit clamp, to scavenge the input signed `32-bit` values.

In the end, I think it's simpler and easier to just clamp the floating-point values to [0,1] (via the `if` statements), and then manage the conversion of _that_ range to [0,255] (through the division into 256 segments and subsequent implicit $\lfloor{x}\rfloor$ function.